src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullServiceImpl.java

```diff
@@ -1,34 +1,135 @@
-        AbstractServiceImpl<SmallFull, SmallFullEntity> implements SmallFullService {
+public class SmallFullServiceImpl implements
+        SmallFullService {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
+
+    private SmallFullRepository smallFullRepository;
+
+    private SmallFullTransformer smallFullTransformer;
+
+    private CompanyAccountService companyAccountService;
+
+    private MessageDigest messageDigest;
 
     @Autowired
-    public SmallFullServiceImpl(
-            @Qualifier("smallFullRepository") MongoRepository<SmallFullEntity, String> mongoRepository,
-            @Qualifier("smallFullTransformer") GenericTransformer<SmallFull, SmallFullEntity> transformer) {
-        super(mongoRepository, transformer);
+    public SmallFullServiceImpl(SmallFullRepository smallFullRepository,
+            SmallFullTransformer smallFullTransformer,
+            CompanyAccountService companyAccountService) {
+        this.smallFullRepository = smallFullRepository;
+        this.smallFullTransformer = smallFullTransformer;
+        this.companyAccountService = companyAccountService;
+    }
+
+    @Override
+    public ResponseObject<SmallFull> create(SmallFull smallFull, Transaction transaction,
+            String companyAccountId, String requestId)
+            throws DataException {
+
+        String selfLink = createSelfLink(transaction, companyAccountId);
+        initLinks(smallFull, selfLink);
+        smallFull.setEtag(GenerateEtagUtil.generateEtag());
+        smallFull.setKind(Kind.SMALL_FULL_ACCOUNT.getValue());
+        SmallFullEntity baseEntity = smallFullTransformer.transform(smallFull);
+
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyAccountId);
+
+        try {
+            baseEntity.setId(generateID(companyAccountId));
+            smallFullRepository.insert(baseEntity);
+            companyAccountService.addLink(companyAccountId, LinkType.SMALL_FULL, selfLink);
+        } catch (DuplicateKeyException dke) {
+            LOGGER.errorContext(requestId, dke, debugMap);
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR, null);
+        } catch (MongoException me) {
+            DataException dataException = new DataException(
+                    "Failed to insert " + ResourceName.SMALL_FULL.getName(), me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        return new ResponseObject<>(ResponseStatus.CREATED, smallFull);
     }
 
     @Override
-    public void addKind(SmallFull rest) {
-        rest.setKind(Kind.SMALL_FULL_ACCOUNT.getValue());
+    public ResponseObject<SmallFull> findById(String id) {
+        SmallFullEntity smallFullEntity = smallFullRepository.findById(id).orElse(null);
```

Comment:
as above- findById should be throwing DataException and passing up the chain here